staff/sean-daryanani/pre-course/calculadora/main.js

```diff
@@ -0,0 +1,313 @@
+//Declare global variables
+var firstNumber=0;
+var secondNumber=0;
+var currentOperand;
+var result = 0;
+var currentNumber = 0;
+var operandSelected;
+var calcStatus= {
+    first:true,
+    second:false,
+    result:false,
+    notNumberClicked:false,
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Functions
+
+
+function removeBackground() {
+    for (var i=0; i<numbers.length; i++) {
+        numbers[i].classList.remove("selected");
+        if (operators[i]) {
+            operators[i].classList.remove("selected");
+        }
+    }
+}
+
+
+function makeVisible() {
+    for (var i=0 ; i<operators.length; i++) {
+        operators[i].style.visibility="visible";
+    }
+    plusMinus.style.visibility = 'visible';
+    percent.style.visibility = 'visible';
+
+}
+
+function hideOperators() {
+    if (currentOperand === '+') {
+        operators[0].style.visibility= 'hidden';
+        operators[1].style.visibility= 'hidden';
+        operators[2].style.visibility= 'hidden';
+        plusMinus.style.visibility = 'hidden';
+        percent.style.visibility = 'hidden'
+    }
+    else if (currentOperand === '-') {
+        operators[0].style.visibility= 'hidden';
+        operators[1].style.visibility= 'hidden';
+        operators[3].style.visibility= 'hidden';
+        plusMinus.style.visibility = 'hidden';
+        percent.style.visibility = 'hidden'
+        
+    }
+    else if (currentOperand === '*') {
+        operators[0].style.visibility= 'hidden';
+        operators[2].style.visibility= 'hidden';
+        operators[3].style.visibility= 'hidden';
+        plusMinus.style.visibility = 'hidden';
+        percent.style.visibility = 'hidden'
+    }
+    else if (currentOperand === '/') {
+        operators[1].style.visibility= 'hidden';
+        operators[2].style.visibility= 'hidden';
+        operators[3].style.visibility= 'hidden';
+        plusMinus.style.visibility = 'hidden';
+        percent.style.visibility = 'hidden'
+    }
+}
+
+function equalsLogic() {
+    if (currentOperand === '+') {
+        result = parseFloat(firstNumber)+ parseFloat(secondNumber)
+        input.value = Math.round(result*100000)/100000;
+    }
+
+    else if (currentOperand === '-') {
+        result = parseFloat(firstNumber)- parseFloat(secondNumber)
+        input.value = Math.round(result*100000)/100000;
+    }
+
+    else if (currentOperand === '*') {
+        result = parseFloat(firstNumber)* parseFloat(secondNumber)
+        input.value = Math.round(result*100000)/100000;
+    }
+
+    else if (currentOperand === '/') {
+        result = parseFloat(firstNumber)/ parseFloat(secondNumber)
+        input.value = Math.round(result*100000)/100000;
+        if (result === Infinity || result === -Infinity || result === NaN) {
+            input.value = "Can't divide by 0!"

```

Comment:
It's really cool, but with the operation 0/0 it doesn't work.

Reply:
I tried fixing the bug but can't figure out why it's not working. My if condition includes result===NaN! Maybe you can help me tomorrow